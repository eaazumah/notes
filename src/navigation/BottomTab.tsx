import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import React from 'react';
import Icon from '../@shared/components/Icon';
import {routes as r} from '../@shared/constants';
import useTheme from '../@shared/hooks/useTheme';
import Archived from '../archived/Archived';
import Favorites from '../favorites/Favorites';
import Home from '../home/Home';

const routes = [
  {
    name: r.home,
    component: Home,
  },
  {
    name: r.favorites,
    component: Favorites,
  },
  {
    name: r.archived,
    component: Archived,
  },
];

const Tab = createBottomTabNavigator();

const BottomTab = () => {
  const {theme} = useTheme();
  return (
    <Tab.Navigator screenOptions={{headerShown: false, ...theme.tabBarOptions}}>
      {routes.map((route, index) => (
        <Tab.Screen
          key={index}
          name={route.name}
          component={route.component}
          options={{
            tabBarIcon: () => (
              <Icon size={12} name={'archive'} type={'EvilIcons'} />
            ),
          }}
        />
      ))}
    </Tab.Navigator>
  );
};

export default BottomTab;
