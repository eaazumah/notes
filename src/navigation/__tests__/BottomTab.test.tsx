jest.useFakeTimers();

import { NavigationContainer } from '@react-navigation/native';
import { cleanup, render } from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../__testUtils__/Providers';
import BottomTab from '../BottomTab';

describe('Bottom Tab Navigator', () => {
  afterEach(cleanup);
  test('Bottom Tab Navigator renders correctly', () => {
    const bottomTab = render(
      <TestProvider>
        <NavigationContainer  >
          <BottomTab />
        </NavigationContainer>
      </TestProvider>,
    );
    expect(bottomTab.toJSON()).toMatchSnapshot();
  });
});
