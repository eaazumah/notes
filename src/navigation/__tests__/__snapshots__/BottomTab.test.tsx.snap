// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom Tab Navigator Bottom Tab Navigator renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenContainer
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F5F5F5",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "space-between",
                    "marginTop": 50,
                    "paddingHorizontal": 17,
                    "width": "100%",
                  },
                ]
              }
              testID="header"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#242A37",
                      "fontSize": 30,
                      "fontWeight": "800",
                    },
                  ]
                }
                testID="title"
              >
                Notes
              </Text>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="open-settings"
              >
                <Icon
                  color="#242A37"
                  name="settings"
                  size={25}
                  testID="icon"
                />
              </View>
            </View>
            <RCTScrollView
              ListEmptyComponent={[Function]}
              data={Array []}
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEnabled={false}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              viewabilityConfigCallbackPairs={Array []}
            >
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#F5F5F5",
                        "height": 1134,
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <SvgMock
                    height={200}
                    width={200}
                  />
                </View>
              </View>
            </RCTScrollView>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#3269ff",
                  "borderRadius": 60,
                  "bottom": 15,
                  "height": 60,
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "right": 15,
                  "width": 60,
                }
              }
            >
              <Icon
                color="#FFFFFF"
                name="note"
                size={25}
                testID="icon"
              />
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenContainer>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 49,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Home, tab, 1 of 3"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": true,
          }
        }
        accessibilityStates={
          Array [
            "selected",
          ]
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Icon
              color="#3269ff"
              name="notes"
              size={28}
              testID="icon"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Icon
              color="#242A37"
              name="notes"
              size={28}
              testID="icon"
            />
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Favorites, tab, 2 of 3"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Icon
              color="#3269ff"
              name="bookmark-outline"
              size={28}
              testID="icon"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Icon
              color="#242A37"
              name="bookmark-outline"
              size={28}
              testID="icon"
            />
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Archived, tab, 3 of 3"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Icon
              color="#3269ff"
              name="archive"
              size={33}
              testID="icon"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Icon
              color="#242A37"
              name="archive"
              size={33}
              testID="icon"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
